### PR TITLE
ADEN-5825 Adjust adUnits for VUAP / ABCD VAST requests

### PR DESCRIPTION
**Pull request by v3rth**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/217b73e1c0678a4776a01c0360a572a6a8c47863/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/217b73e1c0678a4776a01c0360a572a6a8c47863/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1157 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/video/vastUrlBuilder.js [hunk 1]
```diff
@@ -47,31 +47,17 @@ define('ext.wikia.adEngine.video.vastUrlBuilder', [
...
-
-	/**
-	 * Creates VAST URL from given parameters.
-	 * If `options.useMegaAdUnitBuilder` is not explicitly provided, the `context.opts.megaAdUnitBuilderEnabled` is used.
-	 * @param {number} aspectRatio
-	 * @param {object} slotParams
-	 * @param {object} options
-	 * @returns {string}
-	 */
 	function build(aspectRatio, slotParams, options) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/video/vastUrlBuilder.js]: MAJOR Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/AdEngine/js/video/vastUrlBuilder.js [hunk 2]
```diff
@@ -47,31 +47,17 @@ define('ext.wikia.adEngine.video.vastUrlBuilder', [
...
 		var correlator = options.correlator || Math.round(Math.random() * 10000000000),
-			useMegaAdUnitBuilder = options.useMegaAdUnitBuilder,
 			params,
 			url;
 
-		if (typeof useMegaAdUnitBuilder === 'undefined') {
-			useMegaAdUnitBuilder = adContext.getContext().opts.megaAdUnitBuilderEnabled;
-			log(['build', '`useMegaAdUnitBuilder` option is not defined. Fallback to `context.opts.megaAdUnitBuilderEnabled`.'], 'debug', logGroup);
+		if (!options.adUnit && adContext.get('opts.megaAdUnitBuilderEnabled')) {
+			// TODO: remove it after merge & release ADEN-5825 mobile-wiki, this is cache related backward compatibility
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/video/vastUrlBuilder.js]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 3 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/fef54fbb9a1585b948799d45e89c202b80a9e0f8/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/fef54fbb9a1585b948799d45e89c202b80a9e0f8/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1157 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/video/vastUrlBuilder.js (hunk 2 above)
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/video/vastUrlBuilder.js]: Let's create a follow-up ticket for it

File: extensions/wikia/AdEngine/js/video/vastUrlBuilder.js (hunk 1 above)
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/video/vastUrlBuilder.js:59]: What can we do about it? Extract the pre-, mid- and post-rolls logic?

File: extensions/wikia/AdEngine/js/context/adContext.js [hunk 3]
```diff
@@ -281,9 +281,22 @@ define('ext.wikia.adEngine.adContext', [
 		callbacks.push(callback);
 	}
 
+	function get(path) {
+		var isInputVariableCorrect = path === undefined || path === '',
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/context/adContext.js]: Nit-picking: how about changing `isInputVariableCorrect` to `isPathValid`? ;) Does it mean path is valid when it's an undefined or an empty string :thinking:

Comment (v3rth) [reply extensions/wikia/AdEngine/js/context/adContext.js]: ...oooopsss, shame :bell:

File: extensions/wikia/AdEngine/js/video/vastUrlBuilder.js [hunk 4]
```diff
@@ -47,40 +47,30 @@ define('ext.wikia.adEngine.video.vastUrlBuilder', [
 		return aspectRatio >= 1 || !isNumeric(aspectRatio) ? adSizes.horizontal : adSizes.vertical;
 	}
 
-	function buildAdUnit(slotParams, useMegaAdUnitBuilder) {
-		var adUnitBuilder = useMegaAdUnitBuilder ? megaAdUnitBuilder : regularAdUnitBuilder;
-		return adUnitBuilder.build(slotParams.pos, slotParams.src);
+	function getAdUnit(options, slotParams) {
+		if (!options.adUnit && adContext.get('opts.megaAdUnitBuilderEnabled')) {
+			// TODO: remove it after merge & release ADEN-5825 mobile-wiki, this is cache related backward compatibility
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/video/vastUrlBuilder.js:52]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/3a7d85184e2597719d2b7a09746982e4d91f5b6b/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/3a7d85184e2597719d2b7a09746982e4d91f5b6b/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1157 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/video/vastUrlBuilder.js (hunk 2 above)
Comment (v3rth) [reply extensions/wikia/AdEngine/js/video/vastUrlBuilder.js]: https://wikia-inc.atlassian.net/browse/ADEN-5864

File: extensions/wikia/AdEngine/js/video/vastUrlBuilder.js (hunk 1 above)
Comment (v3rth) [reply extensions/wikia/AdEngine/js/video/vastUrlBuilder.js:59]: I extracted part of functionality to function

**Review by nandy-andy — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/cda9f63d5857e0ecbeda18283ed6ec9d200828bf/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/cda9f63d5857e0ecbeda18283ed6ec9d200828bf/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1157 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/d754985ae8f8fa28c7ee92a178b507051d2c9f90/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/d754985ae8f8fa28c7ee92a178b507051d2c9f90/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1157 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/6aadfdc9f0e37cc2e16932919132b6f59fd9d1e5/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/6aadfdc9f0e37cc2e16932919132b6f59fd9d1e5/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1157 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**